### DISCUSSION TITLE
Shelly Dimmer 2 - Single Momentary button

### DISCUSSION THREAD

**Discussion by getinmybelly** [RavenSystem/esp-homekit-devices#1817] [MEPLHAA Scripts]
Wondering if anyone could assist me with the JSON config for my shelly dimmer 2? 
I started with the standard JSON published and changed the GPIO to suit my current need, though would like to so some tweaking if at all possible.
My setup is: 1 momentary switch connected to SW1 (GPIO 14).
Ideal configuration would be:
- Once the button is single pressed it turns off / turns on and set to 100% (currently it returns to the previous brightness)
- A bonus would have been to have similar to how the shelly firmware, where if long pressed it increase brightness until you let go, long press again on the same button it would decrease the brightness.
- Lastly if it would switch off if the internal temperature / over current exceed a specified value

Any assistance would be greatly appreciated.

My current JSON:
`{"c":{"r":[{"n":0}],"l":16,"i":0,"o":6,"b":[{"g":13,"t":5}]},"a":[{"0":{"u":[{"v":"010202060000f401000000ff04"}]},"t":30,"b":[{"g":14,"i":1,"p":0,"t":0}],"y0":[{"0":{"u":[{"v":"010202060a00f4010000010904"}]},"v":1},{"0":{"u":[{"v":"010202063200f4010000013104"}]},"v":5},{"0":{"u":[{"v":"010202066400f4010000016304"}]},"v":10},{"0":{"u":[{"v":"010202069600f4010000019504"}]},"v":15},{"0":{"u":[{"v":"01020206c800f401000001c704"}]},"v":20},{"0":{"u":[{"v":"01020206fa00f401000001f904"}]},"v":25},{"0":{"u":[{"v":"010202062c01f4010000012c04"}]},"v":30},{"0":{"u":[{"v":"010202065e01f4010000015e04"}]},"v":35},{"0":{"u":[{"v":"010202069001f4010000019004"}]},"v":40},{"0":{"u":[{"v":"01020206c201f401000001c204"}]},"v":45},{"0":{"u":[{"v":"01020206f401f401000001f404"}]},"v":50},{"0":{"u":[{"v":"010202062602f4010000012704"}]},"v":55},{"0":{"u":[{"v":"010202065802f4010000018b04"}]},"v":60},{"0":{"u":[{"v":"010202068a02f4010000010904"}]},"v":65},{"0":{"u":[{"v":"01020206bc02f401000001bd04"}]},"v":70},{"0":{"u":[{"v":"01020206ee02f401000001ef04"}]},"v":75},{"0":{"u":[{"v":"010202060032f4010000013104"}]},"v":80},{"0":{"u":[{"v":"010202065203f4010000015404"}]},"v":85},{"0":{"u":[{"v":"010202068403f4010000018604"}]},"v":90},{"0":{"u":[{"v":"01020206b603f401000001b804"}]},"v":95},{"0":{"u":[{"v":"01020206e803f401000001ea04"}]},"v":100}]}]}`

**Comment by andyallcorn** [comment]
I may be able to help you with 1 and 3, but I use a toggle switch so 2 is out of my experience...

I actually do a similar thing with brightness for a dimmer I have; I want the light to come on at 100% every time. (Well, actually, I have it come on at 10% during the night instead as it's the landing light; but this simplified version should just do what you ask). I tried setting brightness on switch On, but that had the side effect of it turning on at 100% _even if I used the Home app to specifically set at a lower value_; not great. So, I came up with a method that works for me -- YMMV of course -- whereby every Off event sets to 100% before immediately once again turning Off, via a hidden switch that has inching to avoid getting into an infinite loop. This does work well for me, I hope it might for you.

I've copied the temperature monitoring from the Shelly 2.5 example which in theory has the same method of monitoring. It's available in the HAA Monitor app but hidden from HomeKit, so you can use that to verify that the figure looks believable. At 70ºC, it turns the lightbulb off. (You can set it to whatever temperature you like.)

`{"c":{"r":[{"n":0}],"l":16,"i":0,"o":6,"b":[{"g":13,"t":5}]},"a":[{"0":{"u":[{"v":"010202060000f401000000ff04"}],"m":[[2,-10001],[2,2],[2,1]]},"1":{"m":[[2,-10000]]},"t":30,"b":[{"g":14,"i":1,"p":0,"t":0}],"y0":[{"0":{"u":[{"v":"010202060a00f4010000010904"}]},"v":1},{"0":{"u":[{"v":"010202063200f4010000013104"}]},"v":5},{"0":{"u":[{"v":"010202066400f4010000016304"}]},"v":10},{"0":{"u":[{"v":"010202069600f4010000019504"}]},"v":15},{"0":{"u":[{"v":"01020206c800f401000001c704"}]},"v":20},{"0":{"u":[{"v":"01020206fa00f401000001f904"}]},"v":25},{"0":{"u":[{"v":"010202062c01f4010000012c04"}]},"v":30},{"0":{"u":[{"v":"010202065e01f4010000015e04"}]},"v":35},{"0":{"u":[{"v":"010202069001f4010000019004"}]},"v":40},{"0":{"u":[{"v":"01020206c201f401000001c204"}]},"v":45},{"0":{"u":[{"v":"01020206f401f401000001f404"}]},"v":50},{"0":{"u":[{"v":"010202062602f4010000012704"}]},"v":55},{"0":{"u":[{"v":"010202065802f4010000018b04"}]},"v":60},{"0":{"u":[{"v":"010202068a02f4010000010904"}]},"v":65},{"0":{"u":[{"v":"01020206bc02f401000001bd04"}]},"v":70},{"0":{"u":[{"v":"01020206ee02f401000001ef04"}]},"v":75},{"0":{"u":[{"v":"010202060032f4010000013104"}]},"v":80},{"0":{"u":[{"v":"010202065203f4010000015404"}]},"v":85},{"0":{"u":[{"v":"010202068403f4010000018604"}]},"v":90},{"0":{"u":[{"v":"01020206b603f401000001b804"}]},"v":95},{"0":{"u":[{"v":"01020206e803f401000001ea04"}]},"v":100}],"es":[{"t":22,"h":2,"n":5,"j":10,"y0":[{"v":70,"r":1,"0":{"m":[[1]]}}]}]},{"t":1,"h":0,"ks":0,"i":1,"s":0,"1":{"m":[[1,102],[1]]}}]}`

**Reply by getinmybelly** [reply]
Hello @andyallcorn. 
Thank you so much for your help with this 👍

**Reply by getinmybelly** [reply]
Hello @andyallcorn, I just loaded it up and the temperature service seems to work perfectly. 👍
Strangely the button ON at 100% still not working though as it returns to the previous dim level of the ios set brightness.

**Reply by andyallcorn** [reply]
Ah, I messed up — the messages are sent to the wrong service.

`"0":{"u":[{"v":"010202060000f401000000ff04"}],"m":[[2,-10001],[2,2],[2,1]]},"1":{"m":[[2,-10000]]}`

that should read:

`"0":{"u":[{"v":"010202060000f401000000ff04"}],"m":[[3,-10001],[3,2],[3,1]]},"1":{"m":[[3,-10000]]}`

The service order is different on my device (no temperature sensor), so the hidden switch is service 3 here instead of 2.

**Reply by getinmybelly** [reply]
Thank you so much. Works like a charm now 👍
